### ISSUE TITLE
java.lang.RuntimeException: store result must not be null with jsr107 and xa-store

### ISSUE THREAD

**Issue by kruthoff** [ehcache/ehcache3#1701]
It doesn't work to configure an ehcache with XA transactions and jcache.

Details are explained in the [ehcache-users forum](https://groups.google.com/forum/#!category-topic/ehcache-users/ehcache-core/IuZ-XUYGxM8).

Thanks for the support.

**Comment by ljacomet** [comment]
Needs to be fixed in release/3.1 and all following releases.

**Comment by henri-tremblay** [comment]
Reproduced with

```java
@Test
public void test() throws Exception {
    BitronixTransactionManager transactionManager =
        TransactionManagerServices.getTransactionManager();
    URI uri = getClass().getResource("ehcache.xml").toURI();
    Caching.getCachingProvider().getCacheManager(uri, getClass().getClassLoader());
}
```

```xml
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.ehcache.org/v3" xmlns:jsr107="http://www.ehcache.org/v3/jsr107" xmlns:tx="http://www.ehcache.org/v3/tx" xsi:schemaLocation="http://www.ehcache.org/v3 http://www.ehcache.org/schema/ehcache-core-3.1.xsd http://www.ehcache.org/v3/jsr107 http://www.ehcache.org/schema/ehcache-107-ext-3.1.xsd">
    <service>
        <jsr107:defaults enable-management="false" enable-statistics="true"/>
    </service>
    <service>
        <tx:jta-tm transaction-manager-lookup-class="org.ehcache.transactions.xa.txmgr.btm.BitronixTransactionManagerLookup"/>
    </service>
    <cache alias="mycache">
        <key-type>java.io.Serializable</key-type>
        <value-type>java.io.Serializable</value-type>
        <heap unit="entries">1000</heap>
        <tx:xa-store unique-XAResource-id="xaCache"/>
    </cache>
</config>
```
for 3.1.3 and 3.2.0

**Comment by ljacomet** [comment]
Dues to changes between 3.1 and 3.2, the fix is not that easy in 3.1. As that line is about the be deprecated, fixing only in 3.2 and above.
